Sources/SlidingContainerViewController.swift

```diff
@@ -0,0 +1,400 @@
+
+}
+
+public protocol SlidingContainerDelegate: class {
+
+    func slidingContainerDidBeingDrag(_ slidingContainer: SlidingContainerViewController)
+
+    func slidingContainerDidEndDrag(_ slidingContainer: SlidingContainerViewController)
+
+}
+
+open class SlidingContainerViewController: UIViewController, UIGestureRecognizerDelegate {
+
+    static open let velTreshHold = 4000.0
+
+    @IBOutlet open weak var dragView: UIView?
+
+    @IBOutlet open weak var navView: UIView?
+
+    public weak var presenter: SlidingContainerPresenter!
+
+    public weak var delegate: SlidingContainerDelegate!
+
+    public var upperController: ContainedViewController!
+
+    public var lowerController: ContainedViewController!
+
+    public let upperView: UIView = UIView()
+
+    public let lowerView: UIView = UIView()
+
+    var panDirection: Int = 0
+
+    private var initialPositionSetUp = false
+
+    private var isDragging = false
+
+    private var lastChange = 0.0
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        upperController = presenter.upperController(for: self)
+        lowerController = presenter.lowerController(for: self)
+
+        addChildViewController(upperController.viewController)
+        addChildViewController(lowerController.viewController)
+
+        navView = navView == nil ? view : navView
+
+        let dView = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: 42))
+        dView.backgroundColor = .darkGray
+        dragView = dragView == nil ? dView : dragView
+
+        navView?.addSubview(upperView)
+        navView?.addSubview(dragView!)
+        navView?.addSubview(lowerView)
+
+        upperView.addSubview(upperController.viewController.view)
+        lowerView.addSubview(lowerController.viewController.view)
+
+        lowerController.didMinimizeControllerWith(diff: getMovementDifference())
+        upperController.didMaximizeControllerWith(diff: getMovementDifference())
+
+        lowerController.viewController.didMove(toParentViewController: self)
+        upperController.viewController.didMove(toParentViewController: self)
+
+        let panGesture = UIPanGestureRecognizer(
+            target: self,
+            action: #selector(self.panDragView(sender:))
+        )
+        panGesture.delegate = self
+        panGesture.delaysTouchesBegan = false
+        panGesture.delaysTouchesEnded = false
+        panGesture.cancelsTouchesInView = false
+        navView?.addGestureRecognizer(panGesture)
+    }
+
+    open override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        if !initialPositionSetUp {
+            initialPositionSetUp = true
+            drawViews()
+        }
+        lowerController.viewController.view.frame = frameForLowerController()
+        upperController.viewController.view.frame = frameForUpperController()
+    }
+
+    func getMovementDifference() -> CGFloat {
```

Comment:
I think this can be private/fileprivate, please review the functions defined here and let everything that can be private as private